cmake/setup_custom_targets.cmake

```diff
@@ -60,13 +60,19 @@ FOREACH(_component compat_files documentation examples parameter_gui)
     _add_custom_target(${_component})
   ELSE()
     STRING(TOUPPER ${_component} _componentuppercase)
+
+    SET(_first_message "Error: Could not ${_description_string} disabled component ${_component}.")
+    DECORATE_WITH_STARS(${_first_message} _decorated_first_message)
+    SET(_second_message "Please reconfigure with -DDEAL_II_COMPONENT_${_componentuppercase}=ON")
+    DECORATE_WITH_STARS(${_second_message} _decorated_second_message)
+
     ADD_CUSTOM_TARGET(${_component}
       COMMAND
            ${CMAKE_COMMAND} -E echo ''
         && ${CMAKE_COMMAND} -E echo ''
         && ${CMAKE_COMMAND} -E echo '***************************************************************************'
-        && ${CMAKE_COMMAND} -E echo "**  Error: Could not ${_description_string} disabled component \"${_component}\"."
-        && ${CMAKE_COMMAND} -E echo "**  Please reconfigure with -DDEAL_II_COMPONENT_${_componentuppercase}=ON"
+        && ${CMAKE_COMMAND} -E echo "${_decorated_first_message}"
+        && ${CMAKE_COMMAND} -E echo "${_decorated_second_message}"
```

Comment:
This looks nice, but it isn't particularly easy to read what's happening here. What do others think about the approach to do the formatting in a macro, as opposed to just doing it by hand once when code is written?